src/containers/EventsHandlers/market.ts

```diff
@@ -66,48 +68,76 @@ const MarketEvents = () => {
       return
     }
     const connectEvents = async () => {
-      allPools.forEach((pool, indexPool) => {
-        if (
-          subscribedTick.findIndex(
-            e =>
-              ((e.fromToken === fromToken.toString() && e.toToken === toToken.toString()) ||
-                (e.toToken === fromToken.toString() && e.fromToken === toToken.toString())) &&
-              e.indexPool === indexPool
-          ) !== -1
-        ) {
-          return
-        }
-        subscribedTick.push({
-          fromToken: pool.tokenX.toString(),
-          toToken: toToken.toString(),
-          indexPool
-        })
-        R.forEachObj(poolTicksArray, tick => {
-          tick.forEach(singleTick => {
-            marketProgram
-              .onTickChange(
-                new Pair(pool.tokenX, pool.tokenY, FEE_TIERS[0]),
-                singleTick.index,
-                tickObject => {
-                  dispatch(
-                    actions.updateTicks({
-                      poolIndex: indexPool.toString(),
-                      index: singleTick.index,
-                      tick: tickObject
-                    })
-                  )
-                }
-              )
-              .then(() => {})
-              .catch(() => {})
+      if (tokenFrom && tokenTo) {
+        allPools.forEach((pool, indexPool) => {
+          if (
+            subscribedTick.findIndex(
+              e =>
+                ((e.fromToken === tokenFrom.toString() && e.toToken === tokenTo.toString()) ||
+                  (e.toToken === tokenFrom.toString() && e.fromToken === tokenTo.toString())) &&
+                e.indexPool === indexPool
+            ) !== -1
+          ) {
+            return
+          }
+          subscribedTick.push({
+            fromToken: pool.tokenX.toString(),
```

Comment:
that looks weird, there wouldn't be any danger that fromToken and toToken might have the same value?